app/trace/trace_viewer.tsx

```diff
@@ -74,6 +75,7 @@ export default class TraceViewer extends React.Component<TraceViewProps, {}> {
 
     window.addEventListener("resize", this.onWindowResize);
     window.addEventListener("mousemove", this.onWindowMouseMove);
+    window.addEventListener("click", this.onWindowMouseClick);
```

Comment:
With this approach we'd need to unregister the window-level event listener in `componentWillUnmount`, since React won't do it automatically.

But I think since we only care about clicks inside the canvas, we can register it on the `canvas` element, so the listener will get cleaned up automatically:


```
<canvas onClick={(e) => onCanvasClick(e, panel)} ... />
```

the impl of `onCanvasClick` can also be simplified to

```
const target = panel.getHoveredEvent()?.args?.target;
if (target) {
  router.navigateTo(`?target=${target}#targets`);
}
```

Reply:
Done!